python-client/giskard/testing/tests/statistic.py

```diff
@@ -224,3 +226,36 @@ def test_disparate_impact(model: BaseModel, dataset: Dataset, protected_slicing_
         passed=bool((disparate_impact_score > min_threshold) * (disparate_impact_score < max_threshold)),
         messages=messages,
     )
+
+
+@test(name="Right Label", tags=["heuristic", "classification"])
+def test_independence_chi_square(model: BaseModel, dataset: Dataset,
+                                 slicing_function: SlicingFunction, threshold: float = 0.1) -> TestResult:
+
+    sliced_dataset = dataset.slice(slicing_function)
+    check_slice_not_empty(sliced_dataset=sliced_dataset, dataset_name="dataset",
+                          test_name="test_independence_chi_square")
+    overlapping_array = np.zeros(len(dataset.df))
+
+    original_df = dataset.df.reset_index(drop=True)
+    sliced_df = sliced_dataset.df.reset_index(drop=True)
+    overlapping_idx = original_df[original_df.isin(sliced_df)].dropna().index.values
+    overlapping_array[overlapping_idx] = 1
+
+    if model.meta.model_type == SupportedModelTypes.CLASSIFICATION:
+        from scipy.stats import chi2_contingency
+        predictions = model.predict(dataset).prediction
+        crosstab = pd.crosstab(overlapping_array, predictions)
+        p_value = chi2_contingency(crosstab)[1]
+    elif model.meta.model_type == SupportedModelTypes.REGRESSION:
+        from scipy.stats import ttest_ind
+        predictions = model.predict(dataset).prediction
+        p_value = ttest_ind(overlapping_array, predictions, alternative="two-sided", equal_var=True).pvalue
+
+    passed = p_value < threshold
```

Comment:
I am a bit confused, what are we testing? Because if it’s independence it’s the other way around. `p_value < threshold` indicates the rejection of independence hypothesis.